### ISSUE TITLE
Flag 448-1

### ISSUE THREAD

**Issue by bayboos** [MollyMcEvilley/LSASampleCode#902]
Hello

The LSA_Household table is an aggregate of values from the temporary tlsa_Household sharing the same characteristics. In this case, the relevant characteristics are: HHChronic (a chronic homelessness status across all Enrollments) and ESTChronic (the same value limited to certain project types). The issue seem to be caused by different ways of calculating the values. An example client appears as a HoH in 35 households (35 distinct enrollments) across 8 programs. In some of them, the household is recognized as “HHChronic = 1”, but the majority of them are 0’s. The relevant detail is that all are of the same LSAProjectType = 0.

Now, the query #6.1 populates tlsa_Household with (effectively in this case) one row per HoH’s ID, which means one row for this particular client, with HHChronic being the minimum value across all enrollments - which is 0. However the ESTChronic is populated by query #6.8.1, where the value is being fetched by a tiny sub-query that is also looking for the minimum value across selected enrollments; but with additional condition of HHChronic = 1 - which will always generate a result of “1” when at least one such enrollment can be found.

There are two other example clients in this dataset who are in exactly the same situation, with 10 and 3 enrollments respecively. At this point it seems like a code issue, where the “minimum value across all enrollments” is expected NOT to be smaller than “the minimum value across some enrollments that has to be = 1”.  Unless I’m missing something, the logic behind the flag seems sound while the value-fetching code does not. Any help will be highly appreciated.

Best regards

**Comment by jbloescw** [comment]
I agree that something is wrong with this flag still.
I have a few instances where a HoHID is in an enrollment all alone and other enrollments where the same HoHID has a household member with them that does have a CH status.  Both are legit enrollments but the way CH is being set somewhere along the way seems amiss.
Is this something the customer just leaves a note on the flag file and moves on?

**Comment by TanyaPieterse** [comment]
Hi there,
We are having the same Flag issue, and we have confirmed with out Customer and their Agencies that the client data was entered correctly.  In our scenario:

- Enrollment 1 (RRH Project): Person A (HOH) and Person B (Spouse) are both active participants.  According to the enrollment information the household IS NOT chronically homeless.
- Enrollment 2 (PSH Project): Person A is the HOH and only participant, the enrollment information indicates the enrollment IS NOT considered chronically homeless
- Enrollment 3 (ES NightByBight Project): Person B is the HOH and only participant, the enrollment information indicates the enrollment IS considered chronically homeless

Because Person B is CH in Enrollment 3, and added as a participant to Enrollment 1 - Enrollment 1 gets flagged as CH for HoHID Person A.  But Enrollment 2 is not flagged as CH for HoHID Person A.

For HoHID Person A, the LSA code will set HHChronic to the minimum value and sets HHChronic=0 (step 5.12).  BUT then the Chronic will be 1 for the one Project Type (RRHChronic=1).  This  triggers Flag 448 because HHChronic=0 and ESTChronic+RRHChronic+PSHChronic>0)

Should Step 5.12 instead of looking at the min rather use the max value?  Or if the Code is correct, and this Flag triggers for a client, is it sufficient to just add a note to explain why the data triggered the Flag?  This is set as a Error Flag and not a Warning Flag, so our clients are expecting us to fix it.

Thanks!

**Comment by MollyMcEvilley** [comment]
TLDR:  The flag is correct -- the sample code was setting HHChronic to 0 when it should not have (i.e., when there were both zero and non-zero values for HHChronic in tlsa_HHID).  I have pushed out a fix.

-------------------------------------------------------------------------------------------

@TanyaPieterse  Is Person B under 18 at the time of Enrollment 1?

If yes...

Person B is chronically homeless, which the LSA treats as a personal characteristic based on info from enrollments over time. This is happening correctly in the sample code:
![image](https://user-images.githubusercontent.com/6156815/151388500-53c901ac-dcad-4c38-9fb4-87ecc19498f5.png)

In 5.12, tlsa_HHID.**HHChronic** for Enrollment 1 should be set to 1 (it is set to the minimum of 1, 2, or 3 from HoH/adult household members based on tlsa_Person columns **CHTime**, **CHTimeStatus**, and **DisabilityStatus** or 0 if none of those apply).
![image](https://user-images.githubusercontent.com/6156815/151232055-688596f0-5f3f-4fff-8b89-686a42e02d8f.png)
This is happening correctly:
![image](https://user-images.githubusercontent.com/6156815/151388134-178f1274-d873-4b27-b786-b0fce43ce15f.png)

In 6.1, tlsa.Household.**HHChronic** should be set to 1 (it is set to the minimum of 1,2, or 3 from tlsa_HHID.**HHChronic** where **ActiveHHType** = **HHType** and **HoHIDs** match and 0 if there is no record in tlsa_HHID where **HHChronic** = (1, 2 ,or 3)
![image](https://user-images.githubusercontent.com/6156815/151232750-a3462d63-df65-4270-85e9-d47810a85363.png)
### **_This is NOT happening correctly in the sample code._**
![image](https://user-images.githubusercontent.com/6156815/151389385-23942cd2-c5ce-4bcb-b668-bfd9dc10ef79.png)

In 6.8, tlsa_Household.**RRHChronic** should be set to 1 (it is set to 1 if there is a tlsa_HHID record for RRH where **HHChronic** = 1; otherwise, it's set to 0)
![image](https://user-images.githubusercontent.com/6156815/151233091-0d0566bb-078e-43ba-b532-943081aa00ba.png)
This is happening correctly in the sample code (screenshot above).

**Comment by TanyaPieterse** [comment]
Hi Molly,
Person B's DOB is 10/18/1980 (41 years old) and was entered on these Dates:
Enrollment 1: Entry 09/14/2020, Exit 01/05/2021
Enrollment 3: Entry 04/23/2021, Exit 05/14/2021

This is Person A's info, that is the HOHID of these 2 Projects, where one is flagged as CH because of Person B being CH in Enrollment 3:
![image](https://user-images.githubusercontent.com/44713006/151232703-70e689d7-cde8-420e-ac3b-8f8d4863fac2.png)

Thanks!
